app/models/user.rb

```diff
@@ -43,4 +45,16 @@ class User < ApplicationRecord
   def to_full_name
     "#{first_name} #{last_name}"
   end
+
+  def self.extract_user_ids_from_comment(comment)
+    comment.scan(/@\[.*?\]\((\d+)\)/).flatten.map(&:to_i)
+  end
+
+  def received_celebrate_comments
+    Response.where.not(celebrate_comment: nil).select do |response|
+      user_ids = User.extract_user_ids_from_comment(response.celebrate_comment)
+      puts "User IDs in comment: #{user_ids}"
+      user_ids.include?(id)
+    end
```

Comment:
This is going to be very inefficient unfortunately. I asked ChatGPT4 about it and go this response: https://chat.openai.com/share/7ebfba32-deec-4b7f-b3eb-26522897d6a0

Reply:
The efficiency of this method can be improved by moving some of the filtering logic to the database query instead of loading all responses into memory and performing the filtering in Ruby.

Here's an updated version of the received_celebrate_comments method that utilizes the database query more efficiently:

```
def received_celebrate_comments
  comment_user_ids = User.extract_user_ids_from_comment(celebrate_comment)
  Response.where.not(celebrate_comment: nil)
          .where("? = ANY(STRING_TO_ARRAY(celebrate_comment, ' '))", id)
          .where(id: comment_user_ids)
end
```